### PR TITLE
BaSD-MS-48: implement react final form in psychologists and profiles

### PR DESCRIPTION
**Pull request by GuidoGlielmi**

### CODE REVIEW

File: src/Components/Admin/Profiles/Form/index.js [hunk 1]
```diff
@@ -55,35 +49,53 @@ const Form = () => {
       });
     }
   };
-
+  const required = (value) => (value ? undefined : 'Required');
+  const validate = (formValues) => {
+    const errors = {};
+    if (
+      !formValues.profile?.match(
+        /[A-Za-z ]+/ || formValues.profile?.length < 3 || formValues.profile?.length > 30
```
Comment (nicolasblasco) [inline src/Components/Admin/Profiles/Form/index.js:57]: @GuidoGlielmi length validation isn't working, we should fix this

File: src/Components/Admin/Psychologists/Form/index.js [hunk 2]
```diff
@@ -290,299 +195,319 @@ const Form = () => {
...
+    }
+    if (
+      !formValues.phone?.toString().match(/^\d+$/) ||
+      formValues.phone?.length < 3 ||
+      formValues.phone?.length > 25
+    ) {
+      errors.phone = 'Input should contain between 3 and 25 numbers';
+    }
+    return errors;
+  };
```
Comment (nicolasblasco) [inline src/Components/Admin/Psychologists/Form/index.js:218]: @GuidoGlielmi we should validate address and could also validate password and username. We could follow the same validations made in admins for password and username, and sessions for the address

**Comment by nicolasblasco** [comment]
@GuidoGlielmi we could validate the availability by validating that if a day is checked, the from a to can't be 0. And if it is not checked the from and to have to be 0.

**Review by nicolasblasco — CHANGES_REQUESTED** [review]
@GuidoGlielmi implementation of final form is very good but we should see the comments requested to approve this PR

**Comment by nicolasblasco** [comment]
@GuidoGlielmi username validation isn't working in psychologists component. It should be required

File: src/Components/Admin/Psychologists/Form/index.js [hunk 3]
```diff
@@ -290,299 +195,384 @@ const Form = () => {
       });
     }
   };
+  const required = (value) => {
+    value ? undefined : 'Required';
+  };
```
Comment (nicolasblasco) [inline src/Components/Admin/Psychologists/Form/index.js:200]: @GuidoGlielmi we should change the braces {} for parenthesis for the function to work

**Review by ValentinaCollareta — APPROVED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**